utils/common.py

```diff
@@ -124,20 +125,22 @@ def get_valid_xml_string(string, escape=True):
     return ''
 
 
-def format_date(date):
-    return date and date.strftime('%d-%m-%Y')
+def deep_date_format(date: Union[datetime.date, datetime.datetime], fallback: Union[str, None] = '') -> Union[str, None]:
```

Comment:
You can use `Optional[str]` which is the prefered shorthand for `Union[str, None]`

Reply:
Done.